pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -559,3 +581,20 @@ func (c *Reconciler) checkTimeout(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec, d
 	}
 	return false, nil
 }
+
+func isPodExceedingNodeResources(pod *corev1.Pod) bool {
+	for _, podStatus := range pod.Status.Conditions {
+		if podStatus.Reason == corev1.PodReasonUnschedulable && strings.Contains(podStatus.Message, "Insufficient") {
```

Comment:
The `strings.Contains()` here feels brittle to me but I wasn't sure how else to narrow down an Unschedulable status to one of insufficient resources.

Reply:
i agree! seems a bit unfortunate but i guess that happens sometimes with errors :D